Makefile_packaging.mk

```diff
@@ -0,0 +1,269 @@
+	  if git mailsplit -o$(DEB_TOP)/patches < "$$f" ;then \
+	      fn=$$(basename "$$f"); \
+	      for f1 in $(DEB_TOP)/patches/*;do \
+	        [ -e "$$f1" ] || continue; \
+	        f1n=$$(basename "$$f1"); \
+	        echo "$${fn}_$${f1n}" >> $(DEB_BUILD)/debian/patches/series ; \
+	        mv "$$f1" $(DEB_BUILD)/debian/patches/$${fn}_$${f1n}; \
+	      done; \
+	  else \
+	    fb=$$(basename "$$f"); \
+	    cp "$$f" $(DEB_BUILD)/debian/patches/ ; \
+	    echo "$$fb" >> $(DEB_BUILD)/debian/patches/series ; \
+	    if ! grep -q "^Description:\|^Subject:" "$$f" ;then \
+	      sed -i '1 iSubject: Auto added patch' \
+	        "$(DEB_BUILD)/debian/patches/$$fb" ;fi ; \
+	    if ! grep -q "^Origin:\|^Author:\|^From:" "$$f" ;then \
+	      sed -i '1 iOrigin: other' \
+	        "$(DEB_BUILD)/debian/patches/$$fb" ;fi ; \
+	  fi ; \
+	done
+	touch $@
+
+
+# Move the debian files into the Debian directory.
+ifeq ($(ID_LIKE),debian)
+$(DEB_TOP)/.deb_files : $(shell find debian -type f) \
+	  $(DEB_TOP)/.detar | \
+	  $(DEB_BUILD)/debian/
+	find debian -maxdepth 1 -type f -exec cp '{}' '$(DEB_BUILD)/{}' ';'
+	if [ -e debian/source ]; then \
+	  cp -r debian/source $(DEB_BUILD)/debian; fi
+	if [ -e debian/local ]; then \
+	  cp -r debian/local $(DEB_BUILD)/debian; fi
+	if [ -e debian/examples ]; then \
+	  cp -r debian/examples $(DEB_BUILD)/debian; fi
+	if [ -e debian/upstream ]; then \
+	  cp -r debian/upstream $(DEB_BUILD)/debian; fi
+	if [ -e debian/tests ]; then \
+	  cp -r debian/tests $(DEB_BUILD)/debian; fi
+	rm -f $(DEB_BUILD)/debian/*.ex $(DEB_BUILD)/debian/*.EX
+	rm -f $(DEB_BUILD)/debian/*.orig
+	touch $@
+endif
+
+# see https://stackoverflow.com/questions/2973445/ for why we subst
+# the "rpm" for "%" to effectively turn this into a multiple matching
+# target pattern rule
+$(subst rpm,%,$(RPMS)): $(SPEC) $(SOURCES)
+	rpmbuild -bb $(COMMON_RPM_ARGS) $(RPM_BUILD_OPTIONS) $(SPEC)
+
+$(subst deb,%,$(DEBS)): $(DEB_BUILD).tar.$(SRC_EXT) \
+	  $(DEB_TOP)/.deb_files $(DEB_TOP)/.detar $(DEB_TOP)/.patched
+	rm -f $(DEB_TOP)/*.deb $(DEB_TOP)/*.ddeb $(DEB_TOP)/*.dsc \
+	      $(DEB_TOP)/*.dsc $(DEB_TOP)/*.build* $(DEB_TOP)/*.changes \
+	      $(DEB_TOP)/*.debian.tar.*
+	rm -rf $(DEB_TOP)/*-tmp
+	cd $(DEB_BUILD); debuild --no-lintian -b -us -uc
+	cd $(DEB_BUILD); debuild -- clean
+	git status
+	rm -rf $(DEB_TOP)/$(NAME)-tmp
+	lfile1=$(shell echo $(DEB_TOP)/$(NAME)[0-9]*_$(DEB_VERS)-1_amd64.deb);\
+	  lfile=$$(ls $${lfile1}); \
+	  lfile2=$${lfile##*/}; lname=$${lfile2%%_*}; \
+	  dpkg-deb -R $${lfile} \
+	    $(DEB_TOP)/$(NAME)-tmp; \
+	  if [ -e $(DEB_TOP)/$(NAME)-tmp/DEBIAN/symbols ]; then \
+	    sed 's/$(DEB_RVERS)-1/$(DEB_BVERS)/' \
+	    $(DEB_TOP)/$(NAME)-tmp/DEBIAN/symbols \
+	    > $(DEB_BUILD)/debian/$${lname}.symbols; fi
+	cd $(DEB_BUILD); debuild -us -uc
+	rm $(DEB_BUILD).tar.$(SRC_EXT)
+	for f in $(DEB_TOP)/*.deb; do \
+	  echo $$f; dpkg -c $$f; done
+
+$(SRPM): $(SPEC) $(SOURCES)
+	rpmbuild -bs $(COMMON_RPM_ARGS) $(RPM_BUILD_OPTIONS) $(SPEC)
```

Comment:
I think `$(RPM_BUILD_OPTIONS)` should just get included in `$(COMMON_RPM_ARGS)` at the top of this file as different options could influence the values we get from an `rpm --specfile` (or `rpm --eval` which we don't yet do, but could) query.

Reply:
I was thinking that there may be a case where we have a different option for query than we do for build.

At the time I was trying to get the %include statement to work/

Reply:
I did a test.  It does not work to combine the RPM_BUILD_OPTIONS that are specific to rpmbuild  such as "--with=mysql" to the COMMON_RPM_ARGS.  RPM reports unknown option and does not return the requested data.

Reply:
Ah right.  Yes.  Those are `rpm` commands not `rpmbuild`.